MINI-PROJECTS.md

```diff
@@ -104,6 +104,314 @@ share a bound.
+so possibly some combination of this, trace generation, DRD / Helgrind and then
+refinement of which traces to pick? Perhaps this is even something ACDL like?
+
+### CORAL / Q
+
+Build something similar to the Q tool in SLAM. (?)
+
+### CryptoMinisat
+
+Try [CryptoMinisat](https://github.com/msoos/cryptominisat) vs Minisat.
+
+### k-liveness
+
+Implement [k-liveness](http://www.cs.utexas.edu/~hunt/FMCAD/FMCAD12/013.pdf)
+for software-liveness checking.
+
+Hongyi is now looking at this (?)
+
+### Known Bugs on the Web
+
+Turn the list of `KNOWNBUG`S from the CBMC regressions into a webpage of known bugs
+of the CBMC `develop` branch and the last CBMC release.
+
+### Coding Standards Checker
+
+Build instrumenters for various coding standards, e.g. AutoSar, [MISRA-C](https://en.wikipedia.org/wiki/MISRA_C),
+CERT-C, [HIC++](https://www.perforce.com/resources/qac/high-integrity-cpp-coding-standard), etc.
+
+### Front-End Testing
+
+Run the regressions mentioned [here](https://github.com/kframework/c-semantics/tree/master/tests) and
+the [GCC C Torture tests](https://gcc.gnu.org/onlinedocs/gccint/C-Tests.html) with `goto-cc` and a
+`goto-program` interpreter (mild dependency on the interpreter-related task above).
+
+### Unspecified C Behaviours
+
+Build instrumentation to check for unspecified behaviours in C, as per Section J.2
+(and maybe also J.3) of the C standard. Some of them may be possible to handle
+entirely in the front-end (for example, catching `x = x++` as modifying a variable
+twice between sequence points could be done in side effect removal in ansi-c), while
+others may be best to be instrumentation (for example handling the unspecified nature
+of the order of evaluation of function arguments).
+
+Talk to Michael Tautschnig or Martin Brain for this item.
+
+### Reporting Software Errors with CWE Information
+
+The [Common Weakness Enumeration project](http://cwe.mitre.org/) seeks to classify software
+weaknesses and errors in a common dictionary. Reporting CWE information together with failed
+assertions, such as [Free of Pointer not at Start of Buffer](http://cwe.mitre.org/data/definitions/761.html),
+would help both users of the tools as well as industrial partners. In the first instance, errors
+already caught by present instrumentation should be augmented with additional information. In a
+second step, instrumentation for other problems listed in the CWE database should be developed.
+
+### Machine Learn Refinement settings
+
+The refinement option enables a tactic of under and over approximation to try to solve the
+generated formulae more quickly. There are a large number of possible options – which operations
+to approximate, how much to under or over approximate them, how much to add back if this doesn't
+work, etc. Rather than trying to manually pick the best strategy from these, it would be good
+to machine learn them.
+
+It would probably be worth asking Nando de Freitas' advice on the machine learning side of things.
```

Comment:
Although I am sure that he has many insightful and helpful things to say on the subject, the specific link between him and the bulk of the CPROVER developers (being at the same institution, one floor different, etc.) has gone so maybe we shouldn't explicitly name check him here?